### ISSUE TITLE
ScalingConfiguration when engine is not Aurora is an invalid property

### ISSUE THREAD

**Issue by atkinsonm** [aws-cloudformation/cfn-python-lint#1335]
*cfn-lint version: (`cfn-lint --version`)* 0.26.2

*Description of issue.*

Sample CloudFormation error on `Create|UpdateStack`:

You can't specify scaling configuration for non Aurora Serverless cluster. (Service: AmazonRDS; Status Code: 400; Error Code: InvalidParameterValue; Request ID: bff47f04-6bbc-492c-a74a-b46b1988f957)

It would be nice for cfn-lint to surface this attribute error so that it can be caught before the  `Create|UpdateStack` action.

**Comment by chrisoverzero** [comment]
In `cfn-lint --version` 0.28.0, this test appears to be firing false failures:

```yaml
Database:
  Type: AWS::RDS::DBCluster
  Properties:
    BackupRetentionPeriod: 30
    DBClusterParameterGroupName: default.aurora-postgresql10
    DBSubnetGroupName: !Ref DatabaseSubnetGroup
    EnableHttpEndpoint: true
    Engine: aurora-postgresql
    EngineMode: serverless
    EngineVersion: '10.7'
    KmsKeyId: !Ref EncryptionKey
    MasterUsername: !Sub '{{resolve:secretsmanager:${DatabaseSecret}::username}}'
    MasterUserPassword: !Sub '{{resolve:secretsmanager:${DatabaseSecret}::password}}'
    ScalingConfiguration:
      AutoPause: false
      MaxCapacity: 4
      MinCapacity: 2
    StorageEncrypted: true
```

This resource produces E3028.

**Comment by kddejong** [comment]
@chrisoverzero I think we resolved the E3028 error in this template.

**Comment by kddejong** [comment]
Rule `E3028` solves this issue
